src/libmeasurement_kit/ooni/web_connectivity.cpp

```diff
@@ -527,8 +527,6 @@ static void experiment_dns_query(
             for (auto answer : message->answers) {
                 if (answer.ipv4 != "") {
                     addresses.push_back(answer.ipv4);
-                } else if (answer.hostname != "") {
-                    addresses.push_back(answer.ipv4);
```

Comment:
Note: this caused passing the `CNAME` to the backend to connect to. The result was a failure.

Reply:
That's interesting. I should probably look into this as it's probably a bug in ooni-backend. Filed: https://github.com/TheTorProject/ooni-backend/issues/110

Reply:
Well, to be more specific: the backend said it could not connect to and the JSON contained a `null` where a boolean was expected. That triggered a `nlohmann::json` exception.